.github/workflows/promote.yml

```diff
@@ -5,92 +5,181 @@ on:
+          SVC_CLI_BOT_GITHUB_TOKEN: ${{ secrets.SVC_CLI_BOT_GITHUB_TOKEN}}
+      - name: Check open-ctc status
+        if: always()
+        run: |
+          RESULT=${{ steps.open-ctc.outcome }}
+          if [[ "$RESULT" != 'success' && "$RESULT" != 'skipped' ]]; then
+            echo "Step 'open-ctc' failed! Exiting."
+            exit 1
+          else
+            echo "Step 'open-ctc' was successful or was skipped, proceeding."
+          fi
+
+  # TODO: Add docker-promote
 
   npm-promote:
-    needs: [ctc-open]
-    runs-on:
-      ubuntu-latest
-      # if you try to use yarn here, it will attempt to use the wrong registry and throw 401s
+    needs: [open-ctc-or-skip, get-package-info]
+    if: success()
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/setup-node@v3
         with:
-          node-version: latest
+          node-version: lts/*
+      # NOTE: If you try to use yarn here, it will use the wrong registry and throw 401s
       - run: |
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc
-          npm dist-tag add @salesforce/cli@${{ inputs.version }} ${{ inputs.tag }}
+          npm dist-tag add @salesforce/cli@${{ needs.get-package-info.outputs.version }} ${{ inputs.new-channel }}
 
-  channel-promote:
-    needs: [ctc-open]
+  oclif-promote:
+    needs: [open-ctc-or-skip, get-package-info, build-s3-channel]
+    if: success()
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: latest
+          node-version: lts/*
           cache: yarn
       - run: yarn install
-      # to get the SHA
-      - uses: salesforcecli/github-workflows/.github/actions/versionInfo@main
-        id: version-info
-        with:
-          version: ${{ inputs.version }}
-          npmPackage: '@salesforce/cli'
-      - run: yarn oclif promote --version ${{ steps.version-info.outputs.version }} --sha ${{ steps.version-info.outputs.sha }} --channel ${{ inputs.channel }} --max-age 300 --macos --win --indexes --xz
+      - run: yarn promote --version ${{ needs.get-package-info.outputs.version }} --sha ${{ needs.get-package-info.outputs.sha }} --channel ${{ needs.build-s3-channel.outputs.s3-new-channel }} --max-age 300 --macos --win --indexes --xz
         env:
-          AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
-          AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
 
   # This job is renames the sfdx-x64.pkg that oclif v3 produces to sfdx.pkg which matches what oclif v2 produces.
   # Doing so avoids breaking changes for customers relying on the old URLs to download the CLI.
   # It can be removed once the developer site is linking to the new sfdx-x64.pkg file and the mac signing job is signing the new -x64.pkg file.
   rename-mac-pkg:
```

Comment:
maybe we don't need this at all?  `sfdx` did to preserve backward compatibility.

Reply:
I wondered this too? I didn't know enough about it to remove it. It does seem very `sfdx` focused. We dont currently link to the `sf` pkg on the developer website right?